drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -967,6 +973,144 @@ static int siwx91x_dev_init(const struct device *dev)
+	}
+
+	if (params->setup.responder) {
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	/* implicit -> won't do renegotiation
+	 * explicit -> must do renegotiation for each session
+	 */
+	if (params->setup.implicit) {
+		twt_req->implicit_twt = 1;
+	} else {
+		/* explicit twt is not supported */
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	twt_req->twt_enable = 1;
+	twt_req->wake_int_exp = params->setup.twt_exponent;
+	twt_req->triggered_twt = params->setup.trigger;
+	twt_req->un_announced_twt = !params->setup.announce;
+	twt_req->wake_int_mantissa = params->setup.twt_mantissa;
+	twt_req->wake_duration_unit = 0;
+	twt_req->wake_duration = params->setup.twt_wake_interval;
+
+	if (params->setup.twt_wake_interval > 255 * 256) {
+		twt_req->wake_duration_unit = 1;
+		twt_req->wake_duration = params->setup.twt_wake_interval / 256;
+	} else {
+		twt_req->wake_duration_unit = 0;
+		twt_req->wake_duration = params->setup.twt_wake_interval / 1024;
+	}
+
+	status = sl_wifi_enable_target_wake_time(twt_req);
+	if (status != SL_STATUS_OK) {
+		params->fail_reason = WIFI_TWT_FAIL_CMD_EXEC_FAIL;
+		params->resp_status = WIFI_TWT_RESP_NOT_RECEIVED;
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int siwx91x_set_twt_teardown(struct wifi_twt_params *params, sl_wifi_twt_request_t *twt_req)
+{
+	sl_status_t status = SL_STATUS_OK;
+
+	twt_req->twt_enable = 0;
+
+	if (params->teardown.teardown_all) {
+		twt_req->twt_flow_id = 0xFF;
+	}
+
+	status = sl_wifi_disable_target_wake_time(twt_req);
+	if (status != SL_STATUS_OK) {
+		params->fail_reason = WIFI_TWT_FAIL_CMD_EXEC_FAIL;
+		params->teardown_status = WIFI_TWT_TEARDOWN_FAILED;
+		return -EINVAL;
+	}
+
+	params->teardown_status = WIFI_TWT_TEARDOWN_SUCCESS;
+
+	return 0;
+}
+
+static int siwx91x_set_twt(const struct device *dev, struct wifi_twt_params *params)
+{
+	sl_wifi_twt_request_t twt_req = { };
+	struct siwx91x_dev *sidev = dev->data;
+	int ret;
+
+	__ASSERT(params, "params cannot be a NULL");
+
+	sl_wifi_interface_t interface = sl_wifi_get_default_interface();
+
+	if (FIELD_GET(SIWX91X_INTERFACE_MASK, interface) != SL_WIFI_CLIENT_INTERFACE) {
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	if (sidev->state != WIFI_STATE_DISCONNECTED && sidev->state != WIFI_STATE_INACTIVE &&
+	    sidev->state != WIFI_STATE_COMPLETED) {
+		LOG_ERR("Command given in invalid state");
+		return -EBUSY;
+	}
+
+	if (params->negotiation_type != WIFI_TWT_INDIVIDUAL) {
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	ret = siwx91x_convert_z_sl_twt_req_type(params->setup_cmd);
+	if (ret < 0) {
+		params->fail_reason = WIFI_TWT_FAIL_CMD_EXEC_FAIL;
+		return ret;
+	}
+
+	twt_req.req_type = ret;
+	/*  Device supports Individual TWT only */
+	twt_req.twt_flow_id = params->flow_id;
+	twt_req.negotiation_type = 0;
+
+	if (params->operation == WIFI_TWT_SETUP) {
```

Comment:
Now, you can simplify that:
```
    switch (params->operation) {
    case WIFI_TWT_SETUP:
        return siwx91x_set_twt_setup(params, &twt_req);
    case WIFI_TWT_TEARDOWN:
        return siwx91x_set_twt_teardown(params, &twt_req);
    default:
       return -ENOTSUP;
    }
```
(You can replace `switch ... case` by` if (...) { } else { }` if you prefer)